### PR TITLE
[Integration Tests] Default NotInIBD on all runners and MockIBD always return false

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs [hunk 1]
```diff
@@ -181,13 +172,25 @@ public INetworkPeer CreateNetworkPeerClient()
             var ibdState = new Mock<IInitialBlockDownloadState>();
             ibdState.Setup(x => x.IsInitialBlockDownload()).Returns(() => true);
 
+            ConnectionManagerSettings connectionManagerSettings = null;
```
[the remark this answers is not in the document]
Comment (fassadlr) [reply src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs]: This was the fix for the network folders that were created in %AppData%/StratisNode whilst running integration tests. I mentioned yesterday that I would include it in this PR 👍

**Comment by fassadlr** [comment]
> 
> 
> What about the tests where we want to be in IBD? Like mining etc?

I removed NotInIBD from everywhere and all the tests still passed.. hmmmm Let me check if there any specific ones I missed...

File: src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bokobza) [reply src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/CoreNode.cs]: ah ok cool

**Comment by bokobza** [comment]
yeah but then when you removed NotInIBD, the ones that were implied InIBD will also get isIBD = false.

**Comment by fassadlr** [comment]
> 
> 
> yeah but then when you removed NotInIBD, the ones that were implied InIBD will also get isIBD = false.

OK I've identified 7 tests that used IBD, I will check all of them now.

**Comment by fassadlr** [comment]
@bokobza I've identified these tests which was NOT set up with `NotInIBD`. I added it to the tests and it made no difference to the outcome.

`a_proof_of_stake_node_with_api_enabled`
`TxMempoolMapOrphans`
`MiningAndPropagatingPOW_MineBlockNotPushedToConsensusCode_SupercededByBetterBlockOnReorg_InitialBlockRejected`
`TestSegwit_MinedOnCore_ActivatedOn_StratisNode`
`SmartContracts_AddToMempool_Success`
`SmartContracts_AddToMempool_OnlyValid`
All `PoA.IntegrationTests`

**Comment by bokobza** [comment]
Nice, thanks mate. @noescape00 can you please check if these tests need to be in IBD or not?

**Comment by fassadlr** [comment]
@noescape00, Jeremy is referring to:

a_proof_of_stake_node_with_api_enabled
TxMempoolMapOrphans
MiningAndPropagatingPOW_MineBlockNotPushedToConsensusCode_SupercededByBetterBlockOnReorg_InitialBlockRejected
TestSegwit_MinedOnCore_ActivatedOn_StratisNode
SmartContracts_AddToMempool_Success
SmartContracts_AddToMempool_OnlyValid
All PoA.IntegrationTests

**Comment by noescape00** [comment]
No, those tests can be with IBD disabled, it's ok. 
I think only tests that test logic that is different in IBD comparing to normal running require absence of IBD mocking.
Such tests are about node not staking before fully synced, inability to sync from IBD node, absence of new blocks advertising and stuff like that.

**Pull request merged**